model/main.py

```diff
@@ -0,0 +1,304 @@
+    """
+
+    calibration_data = dataset[~dataset[game_type_header].str.contains(
+            validation_game_substr)]
+
+    validation_data = dataset[dataset[game_type_header].str.contains(
+            validation_game_substr)]
+
+    train_X, train_y = split_features_label(
+            calibration_data, feature_headers, label_header)
+
+    test_X, test_y = split_features_label(
+            validation_data, feature_headers, label_header)
+
+    return train_X, train_y, test_X, test_y
+
+
+def run_subject_model_experiment(
+        compiled_data, feature_headers, label_header, testable_subjects,
+        results_path="results/model/subject-training.csv",
+        subject_header="subject", early_stop_patience=100, verbose=0):
+    """Runs "Subject Model" experiment.
+
+    The training process is as follows: First, a generic model is
+    obtained by training with data from all subjects, removing examples
+    were the game is "Mario". Second, the generic model is specialized
+    using subject-specific data. Lastly, the validation is made with
+    "Mario" data from particular subjects (each model is validated
+    individually).
+
+    Args:
+        compiled_data (pandas.DataFrame): Raw dataset.
+        feature_headers (list of str): Features to be used.
+        label_header (str): Label header.
+        testable_subjects (list of int): Ids of subjects that has Mario data.
+        subject_header (str, optional): Subject column name in dataset.
+            Defaults to "subject".
+        early_stop_patience (int, optional): Parameter for EarlyStopping.
+            Defaults to 100.
+        verbose (int, optional): Verbosity level passed to keras.
+            Defaults to 0.
+    """
+
+    dataset = normalize_and_encode(
+            compiled_data, feature_headers, label_header)
+
+    # Split dataset into calibration and validation
+    # data (aka training and testing data)
+    train_X, train_y, test_X, test_y = calibration_validation_split(
+            dataset, feature_headers, label_header)
+
+    early_stop = keras.callbacks.EarlyStopping(patience=early_stop_patience)
+
+    # Creates the generic, unspecialized, model
+    generic_model = create_model(
+            input_size=len(feature_headers),
+            output_size=len(dataset[label_header].unique()))
+    generic_model.fit(
+            train_X, train_y, epochs=1000, batch_size=500,
+            validation_data=(test_X, test_y), verbose=verbose)
+
+    generic_model.save("readminds_trained_models/generic_model")
+
+    # NOTE: Training the generic model every time in the loop
+    # has given an accuracy of 62.02% while using a single training for
+    # the generic model granted 59.76%. We should check if this is due
+    # to keras lack of weight copying (a bug in our setup then) or if
+    # is really a improvement. Also, training only with subject data
+    # (same as in Bevilacqua's work) has reported an accuracy
+    # of 61.95 (61.53)%.
```

Comment:
This values might be not statistically different, in the experiments from issue #25 we'll find that out.